optuna/_optimize.py

```diff
@@ -219,26 +219,31 @@ def _run_trial(
         else:
             state = TrialState.COMPLETE
 
-    study._tell(trial, state, values)
-
-    if state == TrialState.COMPLETE:
-        study._log_completed_trial(trial, cast(List[float], values))
-    elif state == TrialState.PRUNED:
-        _logger.info("Trial {} pruned. {}".format(trial.number, str(func_err)))
-    elif state == TrialState.FAIL:
-        if func_err is not None:
-            _logger.warning(
-                "Trial {} failed because of the following error: {}".format(
-                    trial.number, repr(func_err)
-                ),
-                exc_info=func_err_fail_exc_info,
-            )
-        elif values_conversion_failure_message is not None:
-            _logger.warning(values_conversion_failure_message)
+    try:
+        trial._after_func(state, values)
+    except Exception:
+        raise
```

Comment:
It is more consistent to chain the exception like https://github.com/optuna/optuna/pull/1781.

Reply:
Hm, #1781 fixes exceptions to not silence caught exceptions. Here, we're propagating and not silencing anything. Do you mean to construct an error here and chain with it? I think there are pros and cons but ss can be seen in #1781, we don't do that too often. In my understanding, it'll be more to document and understand from the user perspective.

Reply:
Thanks for the detailed explanation. I understood. We should not construct an error here.